android/src/main/java/com/robinpowered/react/Intercom/IntercomModule.java

```diff
@@ -223,7 +224,29 @@ private UserAttributes convertToUserAttributes(ReadableMap readableMap) {
         Map<String, Object> map = recursivelyDeconstructReadableMap(readableMap);
         UserAttributes.Builder builder = new UserAttributes.Builder();
         for (Map.Entry<String, Object> entry : map.entrySet()) {
-            builder.withCustomAttribute(entry.getKey(), entry.getValue());
+            String key = entry.getKey();
+            Object value = entry.getValue();
+            if (key.equals("email")) {
+                builder.withEmail((String)value);
+            } else if (key.equals("user_id")) {
+                builder.withUserId((String)value);
+            } else if (key.equals("name")) {
+                builder.withName((String)value);
+            } else if (key.equals("phone")) {
+                builder.withPhone((String)value);
+            } else if (key.equals("language_override")) {
+                builder.withLanguageOverride((String)value);
+            } else if (key.equals("signed_up_at")) {
+                Date dateSignedUpAt = new Date((long)value);
+                builder.withSignedUpAt(dateSignedUpAt);
+            } else if (key.equals("unsubscribed_from_emails")) {
+                builder.withUnsubscribedFromEmails((Boolean)value);
+            } else if (key.equals("companies")) {
+                Log.w(TAG, "Not implemented yet");
+                // Note that this parameter is companies for iOS and company for Android
+            }
+
+            builder.withCustomAttribute(key, value);
```

Comment:
This adds every attribute (including email, user_id etc.) as a customer attribute.

As I mentioned above, custom attributes should be its own top level item. This line could be replaced with:
```
...
} else if (key.equals("custom_attributes")) {
    // value should be a Map here
    builder.withCustomAttributes(value);
}
```

Reply:
Two reasons for this. I need your opinion on this. I wasn't aware of the existence of `withCustomAttribtes` method. Thanks.

1. My team initially modify in this way, however, we also thought that we need to care about backward compatibility for this library. If it's okay to break the backward compatibility, I would love to change that way. This generally happens in RN wrapper libraries when they are coded incorrectly.

2. Can you check if iOS code also works the same (https://github.com/tinycreative/react-native-intercom/blob/8916104d55f351e7ee7a26e027e8470370a072f6/android/src/main/java/com/robinpowered/react/Intercom/IntercomModule.java)? According to what you said, iOS part of this library does not use the custom `userAttributes.customAttributes` which causes the same problem. Or, the iOS and Android' shape for accessing user attributes are designed differently? 

Before our work, iOS already supports those user attributes, but it does not seem to put other attributes in `customAttribtes`, so we might also need to correct it. Am I right?

Reply:
1. Backwards compatibility should not be preserved here because the current behaviour is broken and will not work. Continuing to set everything as a custom attribute does not give any benefit.
2. The structure for user updates has always been the same for iOS and Android. iOS builds the custom attributes here: https://github.com/tinycreative/react-native-intercom/blob/master/iOS/IntercomUserAttributesBuilder.m#L14. It treats attributes and custom attributes the same as I have proposed here. i.e. `custom_attributes` is in the root of the object.

I hope I'm being clear. Let me know if I can clarify anything.